### PR TITLE
Commerce: One step closer!

### PR DESCRIPTION
**Pull request by zfox23**
There's no need to formally test any of this - we'll write the cumulative test plan when we're ready to enable the feature for all. **None of this PR should change the behavior of master code by default.** This time, I actually mean that.

If you want to test this, make sure you have `"commerce": true,` in your `Interface.json`.

There is still work to be done here. Please see [this Google Doc](https://docs.google.com/document/d/1kcQ7orggHqfJGWaaF3x1n5QJH8-gINzj8sP0zsdT9Jk/) for my todo list.

Additionally, *some of the data across Commerce functionality is fake*, including Certificate contents.

At a high level, this PR includes the following:
1. Refactor of wallet status
2. A new WalletScriptingInterface, which currently holds wallet status and a `buy()` method
3. "Gray security image" bugfix
4. New SortableListModel
5. New "First Time in Purchases" tutorial
6. New flow for first time in purchases/checkout
7. Click anywhere outside username dropdown to close
8. New "My Items" view
9. Purchases view is now sorted correctly
10. Settings handle for Commerce fix
11. Purchase status is no longer fake - this changes both the Recent Activity and Purchases views
12. New "green box" for injected Marketplace JavaScript that directs users to create a wallet

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3767/HighFidelity-Beta-PR11495-bbbce964075bdf3bdf6c42f53d68941e2e07935a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3767/HighFidelity-Beta-PR11495-bbbce964075bdf3bdf6c42f53d68941e2e07935a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3771/HighFidelity-Beta-PR11495-255cede808606a4eb58a3bd580d25eb48a90f84f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3771/HighFidelity-Beta-PR11495-255cede808606a4eb58a3bd580d25eb48a90f84f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3773/HighFidelity-Beta-PR11495-1e52e7824bef5eef646247e1a5ab455a07e5fc08.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3773/HighFidelity-Beta-PR11495-1e52e7824bef5eef646247e1a5ab455a07e5fc08.exe

File: interface/resources/qml/hifi/commerce/purchases/Purchases.qml [hunk 1]
```diff
@@ -530,17 +590,43 @@ Rectangle {
...
+
+        for (var i = 0; i < filteredPurchasesModel.count; i++) {
+            filteredPurchasesModel.setProperty(i, "displayedItemCount", itemCountDictionary[filteredPurchasesModel.get(i).id]);
+        }
+    }
+
+    function sortByDate() {
+        filteredPurchasesModel.sortColumnName = "purchase_date";
+        filteredPurchasesModel.isSortingDescending = false;
+        filteredPurchasesModel.quickSort();
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/commerce/purchases/Purchases.qml]: Why not use javascript's built in sort?

Comment (zfox23) [reply interface/resources/qml/hifi/commerce/purchases/Purchases.qml]: I don't think `sort()` lets me sort an object by key, does it?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3796/HighFidelity-Beta-PR11495-1ec0f5d9a0883fbb71def614eba76198c0ad0e87.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3796/HighFidelity-Beta-PR11495-1ec0f5d9a0883fbb71def614eba76198c0ad0e87.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3800/HighFidelity-Beta-PR11495-2b03e08c968fc2dcfd60dd7e90878f8aa9cc335d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3800/HighFidelity-Beta-PR11495-2b03e08c968fc2dcfd60dd7e90878f8aa9cc335d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3829/HighFidelity-Beta-PR11495-11d0062104698608d0ffe923e7d480b79b89e362.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3829/HighFidelity-Beta-PR11495-11d0062104698608d0ffe923e7d480b79b89e362.exe

File: interface/resources/qml/hifi/commerce/purchases/Purchases.qml (hunk 1 above)
Comment (zfox23) [reply interface/resources/qml/hifi/commerce/purchases/Purchases.qml]: Will change :)

**Comment by howard-stearns** [comment]
approving, but decide what you want to do about the sorting.

**Comment by davidkelly** [comment]
One minor thing:  When you setup the wallet, step 4 of 4 in that process prompts you for instructions to backup your keys.  The finish button doesn't come up until you hit that button.  Do we want to force people to hit the button before finishing?  It is fine with me, just checking if that is the spec.

**Comment by zfox23** [comment]
@davidkelly That is indeed the spec! We do want people to hit the button before finishing.

**Comment by davidkelly** [comment]
also, when you buy an item, you have 2 transactions pending.  I understand that, but I'm wondering if we want to collect those up as 2 sides of the same transaction and call that one?  Again, not something I really am worried about, just imagining that it could confuse people maybe?

**Comment by davidkelly** [comment]
in any case I really like the changes a lot

**Comment by zfox23** [comment]
@davidkelly - I could also imagine that confusing people, but I talked to Howard about that when I first implemented it and it seems like this is the right thing to do - at least until people complain :)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3847/HighFidelity-Beta-PR11495-e64b0861a1aa9de93c974aec725dad562e90624a.dmg

**Comment by zfox23** [comment]
build this please

**Comment by howard-stearns** [comment]
BTW, the two-transaction-in-history thing will change when we integrate with bitcoin/elements. But since we don't yet know precisely how that will be done, the idea is to stick with this at least until we know, and possibly until we actually integrate.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3848/HighFidelity-Beta-PR11495-e64b0861a1aa9de93c974aec725dad562e90624a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3848/HighFidelity-Beta-PR11495-e64b0861a1aa9de93c974aec725dad562e90624a.exe

**Pull request merged**